tests/tools/commands/test_repository.py

```diff
@@ -128,7 +128,43 @@ def test_info_verbose(pex, pex_tools_env):
     } == {Requirement.parse(req) for req in requests_info["requires_dists"]}
 
 
-def test_extract(pex, pex_tools_env, tmpdir):
+def test_extract_determinism(pex, pex_tools_env, tmpdir):
+    deterministic_date_time = (
+        DETERMINISTIC_DATETIME.year,
+        DETERMINISTIC_DATETIME.month,
+        DETERMINISTIC_DATETIME.day,
+        DETERMINISTIC_DATETIME.hour,
+        DETERMINISTIC_DATETIME.minute,
+        DETERMINISTIC_DATETIME.second,
+    )
+
+    deterministic_dists_dir = os.path.join(str(tmpdir), "deterministic-dists")
+    subprocess.check_call(
+        args=[pex, "repository", "extract", "-f", deterministic_dists_dir], env=pex_tools_env
+    )
+    with open_zip(
+        os.path.join(deterministic_dists_dir, "requests-2.25.1-py2.py3-none-any.whl")
+    ) as zipfile:
+        infolist = zipfile.infolist()
+        assert len(infolist) > 0
+        for info in infolist:
+            assert deterministic_date_time == info.date_time
+
+    non_deterministic_dists_dir = os.path.join(str(tmpdir), "non_deterministic_dists_dir")
+    subprocess.check_call(
+        args=[pex, "repository", "extract", "-f", non_deterministic_dists_dir, "--use-system-time"],
+        env=pex_tools_env,
+    )
+    with open_zip(
+        os.path.join(non_deterministic_dists_dir, "requests-2.25.1-py2.py3-none-any.whl")
+    ) as zipfile:
+        infolist = zipfile.infolist()
+        assert len(infolist) > 0
+        for info in infolist:
+            assert deterministic_date_time != info.date_time
```

Comment:
Thoughts on checking more than this, like using `filecmp` on the `.whl` itself? Perhaps create two PEXes, extract wheels, and verify that every wheel is identical between the two PEXes.

Reply:
I mean, if they aren't, what do you propose is the next step?

Reply:
I don't have any proposal for next steps: the PR title and this test both claim that it's reproducible, so I was trying to point out it would be helpful to back that up by checking the reproducibility is more than the timestamp.

Perhaps instead rename this test to be `test_extract_determinism_timestamp`?

Reply:
It turned out wheel pack is deterministic mod timestamp so the timestamp fix does produce deterministic wheels. Since the thorny question of what to do if that had not been true is dodged, I went ahead and added full-blown determinism tests.